content/en/blog/_posts/2023-11-02-mid-cycle-1.29.md

```diff
@@ -0,0 +1,81 @@
+
+For general information about cloud controller managers, please see the Kubernetes document
+[Cloud Controller Manager](https://kubernetes.io/docs/concepts/architecture/cloud-controller/).
+
+### Removal of `v1beta2` flow control API group
+
+The flowcontrol.apiserver.k8s.io/v1beta2 API version of FlowSchema and PriorityLevelConfiguration will no longer be [served in v1.29](https://kubernetes.io/docs/reference/using-api/deprecation-guide/#v1-29). 
+
+To prepare for this, you can edit your existing manifests and rewrite client software to use the `flowcontrol.apiserver.k8s.io/v1beta3` API version, available since v1.26. All existing persisted objects are accessible via the new API. Notable changes in `flowcontrol.apiserver.k8s.io/v1beta3` include
+the PriorityLevelConfiguration spec.limited.assuredConcurrencyShares field is renamed to spec.limited.nominalConcurrencyShares.
+ 
+
+### Deprecation of `status.nodeInfo.kubeProxyVersion` field of v1.Node
+
+The `.status.kubeProxyVersion` field for Node objects will be [marked as deprecated](https://github.com/kubernetes/enhancements/issues/4004) in v1.29 in preparation for its removal in a future release. This field is not accurate and is set by kubelet, which does not actually know the kube-proxy version, or even if kube-proxy is running.
+
+### Deprecation of the `SkipReadOnlyValidationGCE` feature gate
+
+The `SkipReadOnlyValidationGCE` feature gate will be [deprecated](https://github.com/kubernetes/kubernetes/pull/121083). Previous versions of Kubernetes on Google Cloud required that workloads (for example Deployments, DaemonSets, etc.) which used PersistentDisk volumes were using them in read-only mode. This validation provided very little value at a relatively high host implementation cost, and will no longer be validated. If this is a problem for a specific use-case, you'll be able to the set SkipReadOnlyValidationGCE gate to false so as to re-enable the validation, and then we hope you would file a Kubernetes bug with details. 
+
+    
+## Major Changes for Kubernetes v1.29
+
+
+### Native sidecar containers promoted to beta
+
+Introduced in alpha in Kubernetes v1.28, native sidecare containers will be [beta](https://github.com/kubernetes/enhancements/issues/753) in Kubernetes v1.29. In addition to fixing known issues, improvements include implementing proper termination ordering and allowing sidecar containers to restart during the shutdown of the Pod.
+
+## Looking Ahead
+
+The official list of [API removals](https://kubernetes.io/docs/reference/using-api/deprecation-guide/#v1-29) planned for Kubernetes v1.29 includes:
+
+* The flowcontrol.apiserver.k8s.io/v1beta2 API version of FlowSchema and PriorityLevelConfiguration will no longer be served in v1.29.
```

Comment:
Please incorporate the feedback from https://hackmd.io/@krol/middlecycle-129

This should cover releases further in the future (1.30 onwards)

Reply:
Hi @sftim, adding my question from the hackmd file about this. This page, https://kubernetes.io/docs/reference/using-api/deprecation-guide/#v1-29, doesn't list any more deprecations past 1.29. Can you recommend a place where we can find the future planned deprecations?

Reply:
Nice.

You could write “At the time of writing, there are no deprecated APIs with removals planned for releases past v1.29”. But basically this is resolved.


(this is because [beta APIs are off by default](https://kubernetes.io/blog/2020/08/21/moving-forward-from-beta/))